### ISSUE TITLE
Vertical cursor offset after chunk output

### ISSUE THREAD

**Issue by clanfear** [rstudio/rstudio#5218]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.363 (and 1.2.1335) 
    OS Version      :  Win 10 Pro 1903 18362.267
    R Version       : 3.6.1

### Steps to reproduce the problem

As it is always present on my machines, I have no idea how to reproduce it on a machine not experiencing it. This is what triggers it:

(1) Create or open any R Markdown document.
(2) Run a chunk with output.
(3) Highlight text after the chunk; there is now an arbitrary vertical offset between visible the cursor position and the highlight location.
(4) Change the zoom level to arbitrarily change the offset some more.

### Describe the problem in detail

[Symptoms similar to this post.](https://community.rstudio.com/t/vertical-mouse-position-offset-in-rmarkdown-documents/29861) [Here is a crude recording.](https://www.youtube.com/watch?v=gP7-F5-qkB4&feature=youtu.be) Running chunks in markdown docs which produce any output results in a varying level of vertical cursor offset after the chunk. There is no horizontal offset. Changing RStudio zoom alters the offset but does not fix it; once changing zoom, however, the offset is usually gone until a chunk is run (then it returns). The degree of offset is different for different chunks but not linearly related to the length of output, though it is worse when the output is large images or widgets. It does not appear to be additive across multiple chunks. It is unaffected by swapping between software and hardware rendering and checking/unchecking any rendering options. It occurs on all monitors and when moving RStudio across monitors. Problem exists on all three of my machines:

(1) Intel i5-9600K desktop with nVidia GTX 1070 video.
(2) Dell XPS 9570 Intel i7-8750H laptop with nVidia GTX1050 video.
(3) Lenovo Yoga 2 Pro Intel i7-4510U with onboard video.

It also exists on both the current daily build and the release version, and persists on fresh reinstalls.  This problem is also not recent--I've been experiencing it for about a year, I think, but I can't be sure when it appeared. I've also experienced it on a lab computer on my campus (unfortunately while live coding for students) but have not checked recently to see if it is still an issue.

### Describe the behavior you expected

I expect the visible cursor to align with the position clicks and selection are resolved.

**Comment by kevinushey** [comment]
I was able to reproduce as well with the following document:

`````
---
title: "Untitled"
output: html_document
---

```{r}
summary(mtcars)
```

Here is some text.
Here is some more text.
`````

To reproduce:

1. Run the chunk of code;
2. Try to click on the final line in the document.

The cursor is placed on the line above.

However, I do _not_ see this in the latest release (v1.2.1335) or the preview release (v1.2.1568); only a recent daily build (v1.3.352).

@clanfear, can you reproduce the issue using the aforementioned document with one of the v1.2 builds? Or is there some extra context required?

**Comment by clanfear** [comment]
I moved to daily builds originally in hope it had fixed display issues in release (it did fix those)--but that was possibly a version earlier than current release (1.2.1335). If I use 1.2.1335 since R 3.6.0 or 3.6.1, I am afflicted with a different but known error: if any code is run in a chunk, the console is spammed with "ERROR: option error has NULL value" [as seen here](https://github.com/rstudio/rstudio/issues/4759). However, when I open that version, in a already-open and run markdown doc with cached output, there is a slight cursor offset below a run chunk. I can't run any new code to verify if it is the same problem due to the error--it might go away once code is rerun. I suspect that is the case if it is behaving normally for you in 1.2.1335 but not 1.3.352.

The solution provided for the error spam is moving to the preview RStudio version, which I had tried recently. I do not see either of these issues in 1.2.1568. This issue is thus solved on my end and must exist in some change between 1.2.1568 and the 1.3.352 build. Thanks for the help!
